lib/platform/github/index.ts

```diff
@@ -204,9 +205,26 @@ export async function initRepo({
   [config.repositoryOwner, config.repositoryName] = repository.split('/');
   let repo: GhRepo;
   try {
+    let infoQuery = repoInfoQuery;
+    if (config.isGhe) {
+      const gheHeaderKey = 'x-github-enterprise-version';
+      const gheQueryRes = await githubApi.head('/');
+      const gheHeaders: Record<string, string> = gheQueryRes?.headers || {};
+      const [, gheVersion] =
+        Object.entries(gheHeaders).find(
+          ([k]) => k.toLowerCase() === gheHeaderKey
+        ) ?? [];
+      if (
+        !gheVersion ||
```

Comment:
Maybe instead of this, test that it's valid semver before calling lt, otherwise it will throw.